src/TransferStats.php

```diff
@@ -18,18 +18,18 @@
     private $handlerErrorData;
 
     /**
-     * @param RequestInterface  $request          Request that was sent.
-     * @param ResponseInterface $response         Response received (if any)
-     * @param float|null        $transferTime     Total handler transfer time.
-     * @param mixed             $handlerErrorData Handler error data.
-     * @param array             $handlerStats     Handler specific stats.
+     * @param RequestInterface       $request          Request that was sent.
+     * @param ResponseInterface|null $response         Response received (if any)
+     * @param float|null             $transferTime     Total handler transfer time.
+     * @param mixed                  $handlerErrorData Handler error data.
+     * @param array                  $handlerStats     Handler specific stats.
      */
     public function __construct(
         RequestInterface $request,
         ResponseInterface $response = null,
         $transferTime = null,
         $handlerErrorData = null,
-        $handlerStats = []
+        array $handlerStats = []
```

Comment:
This is fine because the class if final

Reply:
Reverted... resolving.

Reply:
@Nyholm ok.. l am unresolving this let's continue the discussion here

1. As I said even if it was not final it does not actually matter since I can do this:
```
public function __constructor(){
    parent::__constructor('myval');
}
```

One constructor does not have to follow the same signature as the parent constructor.

2. The strong argument here is not about final or not... The main point here is this:

```
$stats = new TransferStats($request, null, null, null, null, null)
$stats->getHandlerStats(); // someone is expecting null here... although the docblock says otherwise...
```

As you said.. better be safe than sorry.